edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -2352,6 +2352,10 @@ public String getAssignmentScoreStringByNameOrId(final String gradebookUid, fina
 			//Don't fail on this exception
 			log.debug("Assessment not found by name", e);
 		}
+		catch (GradebookSecurityException gse) {
+			log.debug("Student has not permission to retrieve score for unreleased assignment", gse);
+			return score = "";
```

Comment:
I looked through getAssignmentScoreString and this should probably return null vs an empty String.